packages/data-provider/src/repository/submittedRepository.ts

```diff
@@ -1,21 +1,31 @@
-import { SQL, and, count, eq } from 'drizzle-orm/sql';
+import { SQL, and, count, eq, inArray, isNull, or, sql } from 'drizzle-orm/sql';
 
 import { NewSubmittedData, SubmittedData, submittedData } from 'data-model';
 import { BaseDependencies } from '../config/config.js';
 import { ServiceUnavailable } from '../utils/errors.js';
-import { BooleanTrueObject, PaginationOptions, SubmittedDataRepository } from '../utils/types.js';
+import { BooleanTrueObject, PaginationOptions, SubmittedDataResponse } from '../utils/types.js';
 
 const repository = (dependencies: BaseDependencies) => {
 	const LOG_MODULE = 'SUBMITTEDDATA_REPOSITORY';
 	const { db, logger } = dependencies;
 
+	// Column name on the database used to build JSONB query
+	const jsonbColumnName = 'data';
+
 	const paginatedColumns: BooleanTrueObject = {
 		entityName: true,
 		data: true,
 		organization: true,
 		isValid: true,
 		systemId: true,
 	};
+
+	// Enable or disable SoftDelete filter
+	// Manually adding softDelete filter as Drizzle currently doesn't support Soft-Delete option
+	// Important: Make sure all Queries contains this filter
+	const softDelete = true;
```

Comment:
same for this (env var)

Reply:
removed this variable as soft delete is required, there is no other option other than soft delete.